dashboard/test/controllers/api/v1/projects/section_projects_controller_test.rb

```diff
@@ -0,0 +1,55 @@
+require 'test_helper'
+
+class Api::V1::Projects::SectionProjectsControllerTest < ActionController::TestCase
+  STUDENT_STORAGE_ID = 11
+
+  setup do
+    @teacher = create :teacher
+    @section = create :section, user: @teacher
+    @student = create :user
+    create :follower, section: @section, student_user: @student
+
+    student_project_value = '{"name":"Bobs App","level":"/projects/applab","createdAt":"2017-01-24T16:41:08.000-08:00",'\
+      '"updatedAt":"2017-01-25T17:48:12.358-08:00"}'
+    student_project = {id: 22, value: student_project_value}
+
+    hidden_project_value = '{"name":"Hidden App","level":"/projects/playlab","createdAt":"2017-01-01T00:00:00.000-08:00",'\
+      '"updatedAt":"2017-01-01T00:00:00.000-08:00","hidden":true}'
+    hidden_project = {id: 33, value: hidden_project_value}
+
+    stub_user_storage_ids = [{id: STUDENT_STORAGE_ID}]
+    stub_storage_apps = [student_project, hidden_project]
+
+    mock_table = mock('mock pegasus table')
+    PEGASUS_DB.stubs(:[]).returns(mock_table)
+    mock_table.stubs(:where).returns(stub_user_storage_ids).
+      then.returns(stub_storage_apps)
+  end
+
+  test 'student cannot access section projects' do
+    sign_in(@student)
+    get :index, params: {section_id: @section.id}
+    assert_equal @response.status, 403
+  end
+
+  test 'teacher can access section projects' do
+    sign_in(@teacher)
+    get :index, params: {section_id: @section.id}
+    assert_equal @response.status, 200
+  end
```

Comment:
Also (according to `ability.rb`) admins can read sections. I'm not sure if this is functionality we care about, but if it is, it might be worth adding a unit test ensuring that admins can view sections (even those that don't belong to them).